### PR TITLE
[Bug]fix: split autoslim different checkpoint has equal model size

### PR DESCRIPTION
**Pull request by Hiwyl**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

autoslim get subnet by split_checkpoints is wrone.

## Modification

```python
module.weight = nn.Parameter(temp_weight.data.clone())
module.bias = nn.Parameter(module.bias.data[:out_channels].clone())
```

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by pppppM** [comment]
Hi @Hiwyl, thank you very much for your pr.

OpenMMLab use [pre-commit](https://pre-commit.com/) hook to avoid issues of code style.
This PR currently fails lint checks.

You can use the following two commands to get started with the pre-commit hook quickly.

```
pip install pre-commit
pre-commit run --all-files
```

The installation of pre-commit may be affected by the network. 
If you are inconvenient to use pre-commit, you can contact us, and my colleagues can directly fix the lint in your branch.

**Comment by Hiwyl** [comment]
> 
Has been updated.

File: mmrazor/models/pruners/structure_pruning.py [hunk 1]
```diff
@@ -844,8 +853,7 @@ def concat_backward_parser(self, grad_fn, module2name, var2module,
         # However, a shared module will be visited more than once during
         # forward, so it is still need to be traced even if it has been
         # visited.
-        if (name in visited and visited[name]
-                and name not in self.shared_module):
+        if name in visited and visited[name] and name not in self.shared_module:
```
Comment (pppppM) [inline mmrazor/models/pruners/structure_pruning.py:856]: This line is too long to pass the lint check.
More details can be found at https://github.com/open-mmlab/mmrazor/runs/7088305507?check_suite_focus=true

**Comment by pppppM** [comment]
LGTM if passed CI

**Review by pppppM — APPROVED** [review]

**Pull request merged**